app/models/miq_expression/relative_datetime.rb

```diff
@@ -37,7 +31,7 @@ def self.normalize(rel_time, tz, mode = "beginning")
       interval = interval.pluralize
 
       if interval == "hours"
-        beginning_or_end_of_hour(value.to_i.hours.ago.in_time_zone(tz), mode)
+        value.to_i.hours.ago.in_time_zone(tz).public_send("#{mode}_of_hour")
```

Comment:
Does it make sense to avoid the `public_send` by checking for mode?
```ruby
  t = value.to_i.hours.ago.in_time_zone(tz)
  (mode == 'beginning') ? t.beginning_of_hour : t.end_of_hour
```